migrations/migration.py

```diff
@@ -0,0 +1,178 @@
+            applied_migration = self.get_previous_migration(current_migration)
+
+        if applied_migration is None:
+            raise ValueError("No valid migration files to run.")
+
+        migration_path = os.path.join(self.migrations_dir, applied_migration)
+        with open(migration_path, "r") as migration_file:
+            migration_code = migration_file.read()
+            try:
+                exec(migration_code)
+                self.update_latest_applied_migration_in_fhir(applied_migration)
+
+            except Exception as e:
+                print(f"Error executing migration {applied_migration}: {e}")
+
+    def get_next_migration(self, current_migration) -> str:
+        """Retrieve the latest."""
+        for current_node, previous_node in self.migration_sequence.items():
+            if previous_node == current_migration:
+                return current_node
+        return None
+
+    def get_previous_migration(self, current_migration) -> str:
+        """Retrieve the latest."""
+        if current_migration in self.migration_sequence:
+            return self.migration_sequence.get(current_migration)
+        else:
+            return None
+
+    ## FHIR MANAGEMENT LOGIC
+    def get_latest_applied_migration_from_fhir(self) -> str:
+        """Retrieve the latest applied migration migration id from FHIR."""
+        # Logic to retrieve the latest applied migration number from FHIR
+        basic = MigrationManager.get_resource()
+        basic = first_in_bundle(basic)
+
+        if basic is None:
+            self.create_applied_migration_manager()
+
+            return None
+
+        manager = MigrationManager(basic)
+        latest_applied_migration = manager.get_migration()
+
+        return latest_applied_migration
+
+    def update_latest_applied_migration_in_fhir(self, latest_applied_migration: str):
+        """Update the latest applied migration id in FHIR."""
+        # Logic to update the latest applied migration number in FHIR
+        basic = MigrationManager.get_resource()
+        basic = first_in_bundle(basic)
+
+        if basic is None:
+            basic = self.create_applied_migration_manager()
+
+        manager = MigrationManager(basic)
+        manager.update_migration(latest_applied_migration)
+
+
+    def create_applied_migration_manager(self, initial_applied_migration: str = None):
+        """Create new FHIR resource to keep track of Migration History."""
+        message = "No Migration History for this repository. Initializing new Migration Manager"
+
+        audit_entry(
+            message,
+            level='info'
+        )
+
+        # Logic to create new Basic resource keeping track of the migration
+        created_time = datetime.now().astimezone().isoformat()
+        m = {
+            'resourceType': 'Basic',
+            'identifier': [{"system": "http://our.migration.system", "value": system_wide_id_or_latest_applied_migration}],
```

Comment:
i think `identifier` is overkill here - duplicate data gets out of sync if we don't need it, lets not.